client/mm/wrapped_core.go

```diff
@@ -0,0 +1,442 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package mm
+
+import (
+	"fmt"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/db"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/calc"
+	"decred.org/dcrdex/dex/order"
+)
+
+// wrappedCore implements the clientCore interface. A separate
+// instance should be created for each bot, and the core functions will behave
+// as if the entire balance of the wallet is the amount that has been reserved
+// for the bot.
+type wrappedCore struct {
+	clientCore
+
+	mm    *MarketMaker
+	botID string
+	core  clientCore
```

Comment:
No, because sometimes the `wrapped_core` needs to call a function with the same name in the actual core. In this case the field needs to be used, otherwise there will be an infinite loop.

Reply:
You can call methods on an embedded field by using the type name.

```Go
mkt, err := c.clientCore.ExchangeMarket(form.Host, form.Base, form.Quote)
```